main.go

```diff
@@ -114,7 +162,13 @@ func resourceManifestCreate(d *schema.ResourceData, m interface{}) error {
 }
 
 func resourceManifestUpdate(d *schema.ResourceData, m interface{}) error {
-	cmd := kubectl(m, "apply", "-f", "-")
+	kubeconfig, cleanup, err := kubeconfigPath(m)
+	defer cleanup()
```

Comment:
When `err != nil`, it's normally not safe to use the other returned arguments. For example, the following http.Client code will panic if an error's returned

```
resp, err := http.Get("http://example.com/")
defer resp.Body.Close()
if err != nil {
	// handle error
}
```

Please switch this to

```
kubeconfig, cleanup, err := kubeconfigPath(m)
if err != nil {
    return err
}
defer cleanup()
```

If you're concerned about cleanup always being called then kubeconfigPath should call it itself if it's going to return an error.